File: 49-group-anagrams/group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> mp;
        vector<vector<string>> ans;
        
        for (const auto& str : strs) {
            string sortedStr = str;
            sort(sortedStr.begin(), sortedStr.end());
            mp[sortedStr].push_back(str);
        }
        
        for (const auto& [sortedStr, anagrams] : mp) {
            ans.push_back(anagrams);
        }
        
        return ans;
    }
};

File: 49-group-anagrams/group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
   HashMap<String,List<String>> h=new HashMap<>();
        int n=strs.length;
    for(int i=0;i<n;i++){
    String s=strs[i];
    char[] c=s.toCharArray();
        Arrays.sort(c);
       String temp = new String(c);
     if(h.containsKey(temp)){
    List<String> list=h.get(temp);
       list.add(s);  
    h.put(temp,list);
     }   
     else{
      List<String> list=new ArrayList<>();
       list.add(s);
         h.put(temp,list);
        }
    }
    List<List<String>> ans=new ArrayList<>();
        ans.addAll(h.values());
        return ans;
      }     
    }